CMSgov/mint-app
# [EASi 3154] links as documents be

# EASI-3154

## Changes and Description

- Update Documents in GQL and DB to include new fields
    - isLink
    - URL 
- Validate the url with strict validation of format 
- Introduce a new mutation to create a linked Document
- Update relevant resolvers regarding getting download link for linked documents

Note, that the end document has no validation. We don't make any scans, we just store the link



## How to test this change
1. Create a model plan in the backend
2. Create a document link . 
3. View the documents on the front end
4. Upload a document on the front end
5.  Ensure both types of documents still work (note that the linked document formatting will be updated in a separate frontend ticket when the feature is released)

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
